ipalib/config.py

```diff
@@ -460,8 +461,17 @@ def _bootstrap(self, **overrides):
             self.context = 'default'
 
         # Set confdir:
+        self.env_confdir = os.environ.get('IPA_CONFDIR')
         if 'confdir' not in self:
-            if self.in_tree:
+            if self.env_confdir is not None:
+                if (not path.isabs(self.env_confdir)
+                        or not path.isdir(self.env_confdir)):
+                    raise ScriptError(
```

Comment:
Please raise a `PublicError` subclass here (add one to `ipalib.errors` if necessary), this is a generic library code not tied to scripts.